src/book/service.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { PrismaService } from 'src/prisma/service';
+import { bookDTO } from './dtos';
+
+@Injectable()
+export class BookService {
+  constructor(private readonly prisma: PrismaService) {}
+  async createBook(bookDTO: bookDTO) {
+    try {
+      return await this.prisma.book.create({
+        data: bookDTO,
+      });
+    } catch (error) {
+      throw new BadRequestException(error);
+    }
+  }
+
+  async createBookByAuthor(bookDTO: bookDTO, authorId: number) {
```

Comment:
In prisma schema, Book and Author are many-to-many relationship, do we need some kind of `authorIds: number[]` ?

Reply:
At the beginning, I think we can go with one to many with book and author (Only one author for a book) @longbuibao @Balleys1900

Reply:
so @Balleys1900 please edit prisma scheme from many to many to one to many with Book and Author